tests/manage/z_cluster/cluster_expansion/conftest.py

```diff
@@ -107,3 +110,40 @@ def factory(ocs_nodes=False, node_count=3, taint_label=None):
         log.info(f"Successfully Tainted nodes {new_nodes} with {taint_label}")
 
     return factory
+
+
+@pytest.fixture
+def add_capacity_setup(add_nodes):
+    """
+    Check that we have the right configurations before we start the test
+    """
+    log.info("Start add capacity setup")
+    osd_pods_before = get_osd_pods()
+    number_of_osd_pods_before = len(osd_pods_before)
+    if number_of_osd_pods_before >= constants.MAX_OSDS:
+        pytest.skip("We have maximum of OSDs in the cluster")
+
+    # If we use vSphere we may need to add more worker nodes
+    # to the cluster before starting the test
+    if (
+        config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM
+        and number_of_osd_pods_before >= 9
+    ):
+        num_of_expected_wnodes = 6
+        wnodes = get_worker_nodes()
+        num_of_wnodes = len(wnodes)
+        logging.info(
+            f"We have {number_of_osd_pods_before} OSDs in the cluster, "
+            f"and {num_of_wnodes} worker nodes in the cluster"
+        )
+        if num_of_wnodes < num_of_expected_wnodes:
+            num_of_wnodes_to_add = num_of_expected_wnodes - num_of_wnodes
+            logging.info(
+                f"Adding more {num_of_wnodes_to_add} worker nodes to the cluster"
+            )
+            add_nodes(ocs_nodes=False, node_count=num_of_wnodes_to_add)
```

Comment:
shouldn't we pass `True` doe `ocs_nodes`?

Reply:
I label all the worker nodes in line 149 `label_nodes(wnodes_not_in_ocs)`.

Reply:
Fixed